Libplanet.RocksDBStore/RocksDBStore.cs

```diff
@@ -359,28 +411,49 @@ public override void PutTransaction<T>(Transaction<T> tx)
             }
 
             byte[] key = TxKey(tx.Id);
-
-            if (!(_txDb.Get(key) is null))
+            if (!(_txIndexDb.Get(key) is null))
             {
                 return;
             }
 
-            _txDb.Put(key, tx.Serialize(true));
+            var timestamp = tx.Timestamp.ToUnixTimeSeconds();
+            string txDbName = $"epoch{(int)timestamp / _txAtEachEpoch}";
+            RocksDBStoreEpoch epoch = new RocksDBStoreEpoch("Transaction", txDbName);
+            _rwTxLock.EnterWriteLock();
+            if (!_txDbCache.TryGetValue(epoch.DbName, out RocksDb txDb))
+            {
+                txDb = RocksDBUtils.OpenRocksDb(_options, TxDbPath(txDbName));
+                _txDbCache.AddOrUpdate(epoch.DbName, txDb);
+            }
+
+            txDb.Put(key, tx.Serialize(true));
+            _txIndexDb.Put(key, epoch.Serialize());
             _txCache.AddOrUpdate(tx.Id, tx);
+            _rwTxLock.ExitWriteLock();
         }
 
         /// <inheritdoc/>
         public override bool DeleteTransaction(TxId txid)
         {
             byte[] key = TxKey(txid);
 
-            if (_txDb.Get(key) is null)
+            if (_txIndexDb.Get(key) is null)
             {
                 return false;
             }
 
+            _rwTxLock.EnterWriteLock();
+            RocksDBStoreEpoch epoch = RocksDBStoreEpoch.Deserialize(_txIndexDb.Get(key));
+            if (!_txDbCache.TryGetValue(epoch.DbName, out RocksDb txDb))
+            {
+                txDb = RocksDBUtils.OpenRocksDb(_options, TxDbPath(epoch.DbName));
+                _txDbCache.AddOrUpdate(epoch.DbName, txDb);
+            }
+
             _txCache.Remove(txid);
-            _txDb.Remove(key);
+            _txIndexDb.Remove(key);
+            txDb.Remove(key);
+            _rwTxLock.ExitWriteLock();
```

Comment:
`_rwTxLock.ExitWriteLock()` should be inside a `finally` block so that it's guaranteed to be called even if an any error happens.